### PR TITLE
Add the SLA APIs

### PR DESCRIPTION
**Pull request by mprahl**
This PR is just to get an initial review from the team before submitting the PR to PDC.

### CODE REVIEW

File: pdc/apps/componentbranch/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2015 Red Hat
```
Comment (ralphbean) [inline pdc/apps/componentbranch/__init__.py]: All copyright statements in this new PR should be `2017`.  My understanding is that this represents the copyright *on the file containing the header*.

Comment (mprahl) [inline pdc/apps/componentbranch/__init__.py]: @ralphbean thanks for the comment. I fixed it in the latest rebase.

**Comment by mprahl** [comment]
I added the unit tests.

File: pdc/apps/componentbranch/tests.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+    gc.save()
+    rct = ReleaseComponentType.objects.filter(name='rpm').first()
+    branch = ComponentBranch(
+        name='2.7', global_component=gc, type=rct, active=True)
+    branch.save()
+    stcb = SLAToComponentBranch(sla=sla, eol='2020-01-01', branch=branch)
+    stcb.save()
+
+
+class SLAAPITestCase(APITestCase):
```
Comment (ralphbean) [inline pdc/apps/componentbranch/tests.py]: I read this as "Slappy Test Case" on first read.  :hand:   :D

**Comment by ralphbean** [comment]
OK, I *think* this looks good to me (it is always hard to tell with django for me, since everything is indirect).

Does it behave as you would expect @mprahl?

File: pdc/apps/componentbranch/serializers.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                type=component_type,
+                global_component=branch_global_component,
+                active=branch_active if branch_active is not None else True
+            )
+
+        sla_name = validated_data['sla']
+        sla = SLA.objects.filter(name=sla_name).first()
+        if not sla:
+            raise serializers.ValidationError(
+                'The specified SLA "{0}" does not exist'.format(sla_name))
```
Comment (sochotnicky) [inline pdc/apps/componentbranch/serializers.py]: I think rest of PDC APIs generally reply with valid/possible values in these scenarios

**Comment by sochotnicky** [comment]
Generally - lgtm. Do we maybe want to add fixtures with basic predefined values?

File: pdc/apps/componentbranch/serializers.py (hunk 3 above)
Comment (mprahl) [inline pdc/apps/componentbranch/serializers.py]: I ended up modifying the error messages to be more consistent with the rest of PDC (i.e. `{'field_name': ['error_msg']}`), but I think adding in the possible values could get too long.

Comment (sochotnicky) [inline pdc/apps/componentbranch/serializers.py]: OK, fair enough. Though I honestly don't think we will have a lot of these types?

**Comment by mprahl** [comment]
Submitted the PR at https://github.com/product-definition-center/product-definition-center/pull/433

**Pull request closed without merging**